modules/mobile/app/models/mobile/v0/adapters/allergy_intolerance.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+module Mobile
+  module V0
+    module Adapters
+      class AllergyIntolerance
+        def parse(allergies_info)
+          Array.wrap(allergies_info).map do |allergy|
+            allergy_info = allergy['resource']
+            Mobile::V0::AllergyIntolerance.new(
+              id: allergy_info['id'],
+              resourceType: allergy_info['resourceType'],
+              type: allergy_info['type'],
+              clinicalStatus: clinical_status(allergy_info['clinicalStatus']),
+              code: code(allergy_info['code']),
+              recordedDate: allergy_info['recordedDate'],
+              patient: patient(allergy_info['patient']),
+              notes: notes(allergy_info['note']),
+              reactions: reactions(allergy_info['reaction'])
+            )
+          end
+        end
+
+        private
+
+        def clinical_status(attributes)
+          values = Array.wrap(attributes['coding'])
+          coding_hash = values.map do |code|
+            {
+              'system' => code['system'],
+              'code' => code['code']
+            }
+          end
+
+          { 'coding' => coding_hash }
+        end
+
+        def code(attributes)
+          values = Array.wrap(attributes['coding'])
+          coding_hash = values.map do |code|
+            {
+              'system' => code['system'],
+              'code' => code['code'],
+              'display' => code['display']
+            }
+          end
+
+          {
+            'coding' => coding_hash,
+            'text' => attributes['text']
+          }
+        end
+
+        def patient(attributes)
+          {
+            'reference' => attributes.dig('patient', 'reference'),
+            'display' => attributes.dig('patient', 'display')
+          }
+        end
+
+        def recorder(attributes)
+          {
+            'reference' => attributes.dig('recorder', 'reference'),
+            'display' => attributes.dig('recorder', 'display')
+          }
+        end
+
+        def notes(attributes)
+          Array.wrap(attributes).map do |note|
+            {
+              'authorReference' => {
+                'reference' => note.dig('authorReference', 'reference'),
+                'display' => note.dig('authorReference', 'display')
+              },
+              'time' => note['time'],
+              'text' => note['text']
+            }
+          end
+        end
+
+        def reactions(attributes) # rubocop:disable Metrics/MethodLength
```

Comment:
I think you could make an argument for avoiding this rubocop exception and breaking this method down into smaller methods. For example, the manifestation_hash setting seems like it could be a method.

Reply:
Good advice, broke it up